web/modules/custom/rdf_entity/src/Entity/Query/Sparql/SparqlCondition.php

```diff
@@ -4,47 +4,785 @@
 
 use Drupal\Core\Entity\Query\ConditionFundamentals;
 use Drupal\Core\Entity\Query\ConditionInterface;
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\rdf_entity\RdfGraphHandler;
+use Drupal\rdf_entity\RdfFieldHandler;
+use EasyRdf\Serialiser\Ntriples;
 
 /**
  * Defines the condition class for the null entity query.
+ *
+ * @todo: Build a ConditionInterface that extends the ConditionInterface below.
  */
 class SparqlCondition extends ConditionFundamentals implements ConditionInterface {
 
   /**
+   * The rdf graph handler service object.
+   *
+   * @var \Drupal\rdf_entity\RdfGraphHandler
+   */
+  protected $graphHandler;
+
+  /**
+   * The rdf mapping handler service object.
+   *
+   * @var \Drupal\rdf_entity\RdfFieldHandler
+   */
+  protected $fieldHandler;
+
+  /**
+   * Provides a map of filter operators to operator options.
+   *
+   * @var array
+   */
+  protected static $filterOperatorMap = [
+    'IN' => ['delimiter' => ' ', 'prefix' => '', 'suffix' => ''],
+    'NOT IN' => ['delimiter' => ', ', 'prefix' => '', 'suffix' => ''],
+    'IS NULL' => ['use_value' => FALSE],
+    'IS NOT NULL' => ['use_value' => FALSE],
+    'CONTAINS' => ['prefix' => 'FILTER(regex(', 'suffix' => ', "i"))'],
+    'LIKE' => ['prefix' => 'FILTER(regex(', 'suffix' => ', "i"))'],
+    'NOT LIKE' => ['prefix' => 'FILTER(!regex(', 'suffix' => ', "i"))'],
+    'EXISTS' => ['prefix' => 'FILTER EXISTS {', 'suffix' => '}'],
+    'NOT EXISTS' => ['prefix' => 'FILTER NOT EXISTS {', 'suffix' => '}'],
+    // @todo This is not starts with but contains...
+    'STARTS_WITH' => ['prefix' => 'FILTER(regex(', 'suffix' => ', "i"))'],
+    '<' => ['prefix' => '', 'suffix' => ''],
+    '>' => ['prefix' => '', 'suffix' => ''],
+    '>=' => ['prefix' => '', 'suffix' => ''],
+    '<=' => ['prefix' => '', 'suffix' => ''],
+  ];
+
+  /**
+   * The operators that require a default triple to be added.
+   *
+   * In SPARQL, some of the SQL conditions might need more than one conditions
```

Comment:
Well, I meant SQL actually. Like, some of the SQL conditions [if you convert them to RQL] they need more than one pattern to be expressed! because we still use the
`$query->condition('field', 4, '<');` but it needs 1 pattern and 1 filter on RQL.